src/RamTransforms.cpp

```diff
@@ -30,6 +30,42 @@
 
 namespace souffle {
 
+bool ExpandFilterTransformer::expandFilters(RamProgram& program) {
+    // flag to determine whether the RAM program has changed
+    bool changed = false;
+
+    std::vector<std::unique_ptr<RamCondition>> conditionList;
+
+    visitDepthFirst(program, [&](const RamQuery& query) {
+        std::function<std::unique_ptr<RamNode>(std::unique_ptr<RamNode>)> filterRewriter =
+                [&](std::unique_ptr<RamNode> node) -> std::unique_ptr<RamNode> {
+            if (const RamFilter* filter = dynamic_cast<RamFilter*>(node.get())) {
+                const RamCondition* condition = &filter->getCondition();
+                conditionList = toConjunctionList(condition);
+                if (conditionList.size() > 1) {
+                    changed = true;
+                    std::vector<std::unique_ptr<RamFilter>> filters;
+                    for (auto iter = conditionList.end(); iter != conditionList.begin(); --iter) {
```

Comment:
This seems to skip the first condition, and use the end iterator as if it pointed to something valid. It might be better to just use a reverse iterator.
```suggestion
                    for (auto iter = conditionList.rbegin(); iter != conditionList.rend(); ++iter) {
```

Reply:
This was causing segmentation faults incidentally enough